conans/client/graph/build_mode.py

```diff
@@ -53,7 +53,10 @@ def forced(self, conan_file, ref):
 
         # Patterns to match, if package matches pattern, build is forced
         for pattern in self.patterns:
-            if fnmatch.fnmatch(ref.name, pattern) or repr(ref) == pattern:
+            is_matching_pattern = fnmatch.fnmatch(repr(ref), pattern)
+            is_exact_match = (repr(ref) == pattern)
+            is_same_component = repr(ref).startswith(pattern)
```

Comment:
Shouldn't this be also fnmatch (the old one actually)? That it means that if I use `--build=p` will work with any ref starting with `p`? Or am I missing something?

Reply:
Add an assert of a parameter without 🌟

Reply:
Line 58 actually checks if the given reference starts with the pattern. So the check will return true in your case with `p`.
I can add this to the unittests if it isn't in there already.

Reply:
Added a test that shows how this is still working. See `BuildModeTest::test_partly_build_force`

Reply:
Added additional checks for parts of the name

Reply:
Removed the feature where `p` will start match `pattern` or `post` as discussed below. Not sure if I understood wrong from your message or what your intention was. Let me know when there's something still missing... So far it looks like all tests pass (added some more tests as well).